src/pages/VerifySettingsPage/VerifySettingsPage.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState, useCallback } from "react";
+import { useDeskproAppEvents, useDeskproAppClient } from "@deskpro/app-sdk";
+import { getAccounts } from "../../api/api";
+import { VerifySettings } from "../../components/VerifySettings";
+import type { FC } from "react";
+import type { Maybe, Settings } from "../../types";
+import type { IAccount } from "../../api/types";
+
+const getError = (err: Maybe<Error>): string => {
+  const defaultError = "Failed to connect to Wrike, settings seem to be invalid";
+  let error;
+
+  if (!err) {
+    error = defaultError;
+  } else {
+    try {
+      const parsedError = JSON.parse(err.message);
+      const parsedMessage = JSON.parse(parsedError.message);
+      error = parsedMessage.errorDescription || defaultError;
+    } catch (e) {
+      error = defaultError;
+    }
+  }
+
+  return error;
+};
+
+const VerifySettingsPage: FC = () => {
+  const { client } = useDeskproAppClient();
+  const [accounts, setAccounts] = useState<Array<IAccount["name"]>>([]);
+  const [settings, setSettings] = useState<Settings>({});
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string|null>(null);
+
+  const onVerifySettings = useCallback(() => {
+    if (!client || !settings?.access_token) {
+      return;
+    }
+
+    setIsLoading(true);
+    setError(null);
+    setAccounts([]);
+
+    return getAccounts(client, settings)
+      .then((res) => {
+        const accounts = (Array.isArray(res.data) && res.data?.length > 0)
```

Comment:
No need to check array length, `map` will work fine with a blank array.

Reply:
simplified as
`const accounts = (Array.isArray(res.data) ? res.data : []).map((account) => account.name)`